autotest/gdrivers/pcidsk.py

```diff
@@ -97,62 +97,70 @@ def test_pcidsk_4():
 # Test writing metadata to a newly created file.
 
 
-def test_pcidsk_5():
+@pytest.fixture()
+def pcidsk_5(tmp_path):
+
+    fname = str(tmp_path / "pcidsk_5.pix")
 
     # Create testing file.
+    gdal.GetDriverByName("PCIDSK").Create(fname, 400, 600, 1, gdal.GDT_Byte)
+
+    return fname
 
-    gdaltest.pcidsk_ds = gdal.GetDriverByName("PCIDSK").Create(
-        "tmp/pcidsk_5.pix", 400, 600, 1, gdal.GDT_Byte
-    )
+
+def test_pcidsk_5(pcidsk_5):
 
     # Write out some metadata to the default and non-default domain and
     # using the set and single methods.
+    pcidsk_ds = gdal.Open(pcidsk_5, gdal.GA_Update)
```

Comment:
Create() + SetMetadata()  is not necessarily the same as Create() + Close() + Open_in_update_mode() + SetMetadata()  (speaking generally. I haven't looked at the details of this particular driver)
Perhaps just combine test_pcidsk_5,6 and 7 in the same function ?